internals/attachments.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from google.appengine.api import images
```

Comment:
The images API is part of the legacy bundled [services](https://cloud.google.com/appengine/docs/standard/python3/services/access). I know we use existing appengine legacy services but this will be the first time we use the images API. Instead of adding another dependency on the legacy bundled service, we should consider using one of the libraries they mention in their migrating off legacy bundle services [document](https://cloud.google.com/appengine/migration-center/standard/services/migrating-services?tab=python#images)

> To resize, convert, and manipulate images, use an image processing library such as [Pillow](https://python-pillow.org/) or a [Python interface](https://imagemagick.org/script/develop.php#python) for ImageMagick.

At first glance, Pillow has a similar [API](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.resize).

Reply:
Good suggestion.  I'll try to follow up with a PR to switch to Pillow and use max-aspect.